chain/network/src/peer/peer_actor.rs

```diff
@@ -844,21 +848,72 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for PeerActor {
                         .map(|port| SocketAddr::new(self.peer_addr.ip(), port)),
                     account_id: None,
                 };
-                self.chain_info = handshake.sender_chain_info.clone();
-                self.peer_manager_wrapper_addr
-                    .send(ActixMessageWrapper::new_without_size(PeerToManagerMsg::RegisterPeer(RegisterPeer {
-                        actor: ctx.address(),
-                        peer_info: peer_info.clone(),
-                        peer_type: self.peer_type,
-                        chain_info: handshake.sender_chain_info.clone(),
+                self.connection_state = Some(Arc::new(ConnectedPeer {
+                    addr: ctx.address(),
+                    peer_info: peer_info.clone(),
+                    initial_chain_info: handshake.sender_chain_info.clone(),
+                    chain_height: AtomicU64::new(handshake.sender_chain_info.height),
+                    partial_edge_info: handshake.partial_edge_info.clone(),
+                    peer_type: self.peer_type,
+                    stats: ArcSwap::new(Arc::new(Stats {
+                        sent_bytes_per_sec: 0,
+                        received_bytes_per_sec: 0,
+                        encoding: None,
+                    })),
+                    last_time_peer_requested: ArcSwap::new(Arc::new(self.clock.now())),
+                    last_time_received_message: ArcSwap::new(Arc::new(self.clock.now())),
+                    connection_established_time: self.clock.now(),
+                    throttle_controller: self.throttle_controller.clone(),
+                    send_accounts_data_demux: demux::Demux::new(
+                        self.network_state.send_accounts_data_rl,
+                    ),
+                }));
+
+                let connection_state = self.connection_state.clone().unwrap();
+                let tracker = self.tracker.clone();
+                let clock = self.clock.clone();
+                let period = self.network_state.config.peer_stats_period.try_into().unwrap();
+                actix::spawn(async move {
+                    loop {
+                        let sent = tracker.lock().sent_bytes.minute_stats(&clock);
+                        let received = tracker.lock().received_bytes.minute_stats(&clock);
+                        // TODO(gprusak): this stuff requires cleanup: only chain_info.height is
+                        // expected to change. Rest of the content of chain_info is not relevant
+                        // after handshake.
+                        connection_state.stats.store(Arc::new(Stats {
+                            received_bytes_per_sec: received.bytes_per_min / 60,
+                            sent_bytes_per_sec: sent.bytes_per_min / 60,
+                            // TODO(gprusak): deprecated, remove together with the Borsh encoding
+                            // support.
+                            encoding: None,
```

Comment:
Why we do even store it in Stats at all? Let's maybe remove the field and just send None for the metric?

Reply:
done